dags/civicrm_email_report_import_pipeline.py

```diff
@@ -0,0 +1,122 @@
+# Note: DagBag.process_file skips files without "airflow" or "DAG" in them
+
+import os
+import logging
+from datetime import timedelta
+import base64
+from ruamel import yaml
+
+from data_pipeline.utils.dags.data_pipeline_dag_utils import create_dag, create_python_task
+from data_pipeline.utils.pipeline_file_io import get_yaml_file_as_dict
+from data_pipeline.civicrm_email_report.civicrm_email_report import iter_email_report
+from data_pipeline.utils.data_store.bq_data_service import (
+    get_distinct_values_from_bq,
+    load_given_json_list_data_from_tempdir_to_bq
+)
+from data_pipeline.civicrm_email_report.get_civicrm_email_report_data_config import (
+    CiviCrmEmailReportDataConfig
+)
+
+LOGGER = logging.getLogger(__name__)
+
+CIVICRM_KEY_SECRET_FILE_PATH_ENV = "CIVICRM_KEY_SECRET_FILE_PATH"
+
+CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME = "CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH"
+
+DEPLOYMENT_ENV_ENV_NAME = "DEPLOYMENT_ENV"
+DEFAULT_DEPLOYMENT_ENV = "ci"
+
+DAG_ID = "Civicrm_Email_Summary_Report_Pipeline"
+CIVICRM_EMAIL_REPORT_DATA_PIPELINE_SCHEDULE_INTERVAL_ENV_NAME = (
+    "CIVICRM_EMAIL_REPORT_DATA_PIPELINE_SCHEDULE_INTERVAL"
+)
+
+
+def get_env_var_or_use_default(env_var_name, default_value=None):
+    return os.getenv(env_var_name, default_value)
+
+
+def get_data_config(**kwargs):
+    config_file_path = get_env_var_or_use_default(
+        CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME, ""
+    )
+    data_config_dictionary = get_yaml_file_as_dict(config_file_path)
+    kwargs["ti"].xcom_push(
+        key="data_config_dict",
+        value=data_config_dictionary
+    )
+
+
+def data_config_from_xcom(context):
+    dag_con = context["ti"]
+    data_config_dict = dag_con.xcom_pull(
+        key="data_config_dict",
+        task_ids="get_data_config"
+    )
+    deployment_env = get_env_var_or_use_default(
+        DEPLOYMENT_ENV_ENV_NAME,
+        DEFAULT_DEPLOYMENT_ENV
+    )
+    data_configuration = CiviCrmEmailReportDataConfig(
+        data_config_dict, deployment_env)
+    LOGGER.info('data_config: %r', data_configuration)
+    return data_configuration
+
+
+def get_civicrm_credential(key_name: str):
+    secret_file = get_env_var_or_use_default(CIVICRM_KEY_SECRET_FILE_PATH_ENV, "")
+    # pylint: disable=consider-using-with
+    data = yaml.safe_load(open(secret_file))
+    if key_name == "api_key":
+        return base64.b64decode(data["data"]["civi_api_key.txt"])
+    return base64.b64decode(data["data"]["civi_site_key.txt"])
+
+
+def civicrm_email_report_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    email_id_list = get_distinct_values_from_bq(
```

Comment:
What does `get_distinct_values_from_bq` return?
I guess it is a pandas `DataFrame`? What are the columns? You could just select that column.

Reply:
column name can  be `0` probably. Because I did not named it. let me check

Reply:
```
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   index   3010 non-null   int64
 1   0       3010 non-null   int64
```

Reply:
used the column name and get the list afterwards.